packages/app/src/cli/services/function/binaries.ts

```diff
@@ -99,18 +100,46 @@ export async function installBinary(bin: DownloadableBinary) {
 
   const url = bin.downloadUrl(process.platform, process.arch)
   outputDebug(`Downloading ${bin.name} ${bin.version} from ${url} to ${bin.path}`)
-  await mkdir(dirname(bin.path))
-  const resp = await fetch(url)
-  if (resp.status !== 200) {
-    throw new Error(`Downloading ${bin.name} failed with status code of ${resp.status}`)
+  const dir = dirname(bin.path)
+  if (!(await fileExists(dir))) {
+    await mkdir(dir)
   }
+  await performActionWithRetryAfterRecovery(
+    async () => {
+      const controller = new AbortController()
+      // Picking 5000 ms as the timeout because Javy is around 13 megabytes, so it
+      // will take around 4 seconds to download on a 25 Mbit connection, and also
+      // be able to retry twice in the 13 seconds we have for a test to complete
+      // before timing out.
```

Comment:
Our users could have slower internet, so we should probably choose a higher timeout 🤔

Reply:
Yeah that makes sense to me. I'm going to remove this timeout.